fakeredis/commands_mixins/hash_mixin.py

```diff
@@ -134,3 +138,136 @@ def hrandfield(self, key: CommandItem, *args: bytes) -> Optional[List[bytes]]:
+        key: CommandItem,
+        field: bytes,
+        expiration: Union[datetime.datetime, datetime.timedelta, int, float],
+        include_ms: bool,
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> HexpireResult:
+        hash_val: Hash = key.value
+        if field not in hash_val:
+            return HexpireResult.NO_SUCH_KEY
+        current_expiration = hash_val._get_expiration(field)
+        if isinstance(expiration, datetime.datetime):
+            final_expiration = expiration.timestamp()
+        elif isinstance(expiration, datetime.timedelta):
+            final_expiration = time.time() + expiration.total_seconds()
+        else:
+            final_expiration = time.time() + expiration
+        if include_ms:
+            final_expiration = float(final_expiration)
+        else:
+            final_expiration = int(final_expiration)
+        if (
+            (nx and current_expiration is not None)
+            or (xx and current_expiration is None)
+            or (gt and final_expiration <= current_expiration)
+            or (lt and final_expiration >= current_expiration)
+        ):
+            return HexpireResult.CONDITION_UNMET
+        return hash_val._set_expiration(field, final_expiration)
+
+    @command(name="HEXPIRE", fixed=(Key, Union[int, datetime.timedelta]), repeat=(bytes,))
+    def hexpire(
+        self,
+        key: CommandItem,
+        seconds: Union[int, datetime.timedelta],
+        *fields: bytes,
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> list[HexpireResult]:
+        return [self._set_key_expiration(key, field, seconds, False, nx, xx, gt, lt) for field in fields]
+
+    @command(name="HPEXPIRE", fixed=(Key, Union[float, datetime.timedelta]), repeat=(bytes,))
+    def hpexpire(
+        self,
+        key: CommandItem,
+        milliseconds: float,
+        *fields: bytes,
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> list[HexpireResult]:
+        return [self._set_key_expiration(key, field, milliseconds, True, nx, xx, gt, lt) for field in fields]
+
+    @command(name="HEXPIREAT", fixed=(Key, Union[int, datetime.datetime]), repeat=(bytes,))
+    def hexpireat(
+        self,
+        key: CommandItem,
+        unix_time_seconds: Union[int, datetime.datetime],
+        *fields: bytes,
+        nx: bool = False,
+        xx: bool = False,
+        gt: bool = False,
+        lt: bool = False,
+    ) -> list[HexpireResult]:
+        return [self._set_key_expiration(key, field, unix_time_seconds, False, nx, xx, gt, lt) for field in fields]
+
+    @command(name="HPEXPIREAT", fixed=(Key, Union[float, datetime.datetime]), repeat=(bytes,))
+    def hpexpireat(
+        self,
+        key: CommandItem,
+        unix_time_milliseconds: float,
+        *fields: bytes,
+        nx: bool = False,
```

Comment:
... Not sure passing the nx/xx/gt/lt arguments would work like this here.
Check `_expireat` in `generic_mixin.py` to see how to extract arguments.